src/android/CodePush.java

```diff
@@ -461,6 +487,8 @@ public void onStart() {
                 handleAppStart();
                 this.markUpdate();
                 this.codePushPackageManager.clearPendingInstall();
+            } else if (codePushReportingManager.hasFailedReport()) {
```

Comment:
(basically, I'm just wondering why this `else if` isn't also attached to the other `if (pendingInstall...` block above as well)